### ISSUE TITLE
"Fail" properly when plan can't/won't/shouldn't be loaded as a service

### ISSUE THREAD

**Issue by qubitrenegade** [habitat-sh/habitat#5507]
Hello,

I would like to see Habitat emit proper error message when a plan can't/won't/shouldn't be run/loaded as a service.

For instance a `hab svc load core/docker` emits the following cryptic error message:

```
Aug 27 00:18:50 charlie hab[15574]: docker.default(SV): Starting service as user=hab, group=hab
Aug 27 00:18:50 charlie hab[15574]: docker.default(SR): Service restart failed: hab-sup(ER)[components/sup/src/error.rs:512:9]: Unknown: Unable to spawn process, No such file or directory (os error 2)
```

The location of the error in the rust code (`components/sup/src/error.rs:512:9`) is useless in this context.  "Unable to spawn process" is speciously helpful.  "No such file or directory (os error 2)" What file doesn't exist?  Based on this, I'd almost think it was `components/sup/src/error.rs` that didn't exist since it's the only file that's named, but the line/column number eliminates that... this is my equivalent "PC LOAD LETTER!!!111" moment.

What I'd like to see is when I `hab svc load plan/without-run-hook` I don't get the:

```
# hab svc load plan/without-run-hook`
The `plan/without-run-hook` service was successfully loaded
```
I'd get:

```
# hab svc load plan/without-run-hook`
The `plan/without-run-hook` cannot be loaded as it has no way to start (aka no run hook).  Also, it's flagged as a binary package.
```

Lacking that, something like:

```
Aug 27 00:18:50 charlie hab[15574]: docker.default(SV): Starting service as user=hab, group=hab
Aug 27 00:18:50 charlie hab[15574]: docker.default(SR): Service restart failed: hab-sup(ER)[components/sup/src/error.rs:512:9]: there is no run hook defined, no way for hab-sup to execute this package as a process.  Are you sure you can load this service?  Check the readme or something...
```

would be immensely more useful.

Anyway, TL;DR:

* Fail hard when trying to load package without run hook
* In lieu of that fix error message the specifies there's no run hook so no way to start instead nonsensical "file not found" error message.

Thanks!

**Comment by christophermaier** [comment]
@qubitrenegade Yup, agreed. We've got this logged already as https://github.com/habitat-sh/habitat/issues/3468

Going to close this out as a duplicate, but thanks for bringing it up!

**Comment by qubitrenegade** [comment]
heh, I somehow knew there was already an issue about this but didn't find 3468...

Thanks! :+1:
